kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/testplugins/SaslInspectionFilter.java

```diff
@@ -0,0 +1,449 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.testplugins;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.CompletionStage;
+
+import org.apache.kafka.common.errors.AuthenticationException;
+import org.apache.kafka.common.errors.SaslAuthenticationException;
+import org.apache.kafka.common.message.RequestHeaderData;
+import org.apache.kafka.common.message.ResponseHeaderData;
+import org.apache.kafka.common.message.SaslAuthenticateRequestData;
+import org.apache.kafka.common.message.SaslAuthenticateResponseData;
+import org.apache.kafka.common.message.SaslHandshakeRequestData;
+import org.apache.kafka.common.message.SaslHandshakeResponseData;
+import org.apache.kafka.common.protocol.Errors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.kroxylicious.proxy.filter.FilterContext;
+import io.kroxylicious.proxy.filter.RequestFilterResult;
+import io.kroxylicious.proxy.filter.ResponseFilterResult;
+import io.kroxylicious.proxy.filter.SaslAuthenticateRequestFilter;
+import io.kroxylicious.proxy.filter.SaslAuthenticateResponseFilter;
+import io.kroxylicious.proxy.filter.SaslHandshakeRequestFilter;
+import io.kroxylicious.proxy.filter.SaslHandshakeResponseFilter;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+public class SaslInspectionFilter
```

Comment:
did you want this to call `FilterContext#clientSaslAuthenticationSuccess` and failure? Just wondering where this filter is heading, I thought it's purpose would be to report the SASL details to the context.

Reply:
> did you want this to call `FilterContext#clientSaslAuthenticationSuccess` and failure?

If by 'this' you mean 'this class' then it already does on lines 415 and 431 :thinking:

Reply:
d'oh, should have checked it out in IDE for find usages